### PR TITLE
Make comments between if & else to look good

### LINKED ISSUE #487: Comment between if/else is formatted strangely
```js
if(x) {
}
// comment
else {
}
```

output:

```js
if (x) {
} else
  // comment
  {
  }
```

### PR DESCRIPTION
**Pull request by vjeux**
While trying to figure out how to handle both MemberExpression comments and IfStatement comments, I ended up doing this one as well... Sorry @yamafaktory :(

The logic is a bit annoying but works.

Fixes #487

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -1,5 +0,0 @@
-Object.keys(items)
```
Comment (vjeux) [inline test.js]: Oops, I accidentally committed this one.

File: tests/comments/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -70,6 +70,94 @@ declare class Foo extends Qux<string> {/* dangling */}
...
+  // ple
+  // lines
+  // existing comment
+  true;
+else if (4) {
+  // okay?
+  // empty with existing comment
+} else {
+  // comment
+}
```
Comment (vjeux) [inline tests/comments/__snapshots__/jsfmt.spec.js.snap]: All those comments are so well aligned :)

**Comment by yamafaktory** [comment]
@vjeux no problem at all (I was busy yesterday evening anyway, so I didn't start refactoring my PR)! Awesome job 👍 (closes #515).

File: src/comments.js [hunk 3]
```diff
@@ -278,6 +280,36 @@ function addTrailingComment(node, comment) {
...
 
+function addBlockStatementFirstComment(node, comment) {
+  if (node.body.length === 0) {
+    addDanglingComment(node, comment);
+  } else {
+    addLeadingComment(node.body[0], comment);
+  }
+}
+
+function handleIfStatementComments(enclosingNode, followingNode, comment) {
```
Comment (jlongster) [inline src/comments.js]: Can you add a comment for this function to document the use case it's solving?

**Pull request merged**